pkg/summarizer/flakiness.go

```diff
@@ -46,6 +48,37 @@ func CalculateHealthiness(grid *state.Grid, analyzer flakinessAnalyzer, startTim
 	return analyzer.GetFlakiness(gridMetrics, minRuns, startTime, endTime, tab)
 }
 
+// CalculateTrend populates the ChangeFromLastInterval fields of each TestInfo by comparing
+// the current flakiness to the flakiness calculated for the last interval. Interval length
+// is a config value that is 7 days by default. The Trend enum defaults to UNKNOWN, so there
+// is no need to explicitly assign UNKNOWN when a test appears in currentHealthiness but not
+// in previousHealthiness.
+func CalculateTrend(currentHealthiness, previousHealthiness *summarypb.HealthinessInfo) {
+	previousFlakiness := map[string]float32{}
+	// Create a map for faster lookup and avoiding repeated iteration through Tests
+	for _, test := range previousHealthiness.Tests {
+		previousFlakiness[test.DisplayName] = test.Flakiness
+	}
+
+	for i, test := range currentHealthiness.Tests {
+		var prevFlakiness float32
+		if value, ok := previousFlakiness[test.DisplayName]; ok {
+			prevFlakiness = value
+		}
+		currentHealthiness.Tests[i].ChangeFromLastInterval = getTrend(test.Flakiness, prevFlakiness)
```

Comment:
nit: this can be simplified to

```
                currentHealthiness.Tests[i].ChangeFromLastInterval = summarypb.None
                if value, ok := previousFlakiness[test.DisplayName]; ok {
			currentHealthiness.Tests[i].ChangeFromLastInterval = getTrend(test.Flakiness, prevFlakiness)
		}
		
```